pkgs/jni/lib/src/jreference.dart

```diff
@@ -25,70 +24,67 @@ extension ProtectedJReference on JReference {
     }
   }
 
-  /// Similar to [reference].
+  /// Similar to [pointer].
   ///
   /// Detaches the finalizer so the underlying pointer will not be deleted.
   JObjectPtr toPointer() {
     setAsReleased();
-    return _reference;
+    return _pointer;
   }
 }
 
-/// A managed JNI global reference.
-///
-/// Uses a [NativeFinalizer] to delete the JNI global reference when finalized.
-abstract class JReference implements Finalizable {
-  static final _finalizer =
-      NativeFinalizer(Jni.env.ptr.ref.DeleteGlobalRef.cast());
-
-  JReference.fromRef(this._reference) {
-    _finalizer.attach(this, _reference, detach: this);
-  }
-
+sealed class JReference {
```

Comment:
Consider `abstract final`. `sealed` promises exhaustiveness, which means adding things to the hierarchy later would be a breaking change.

https://dart.dev/language/modifier-reference

> This potentially enables us to have multiple kinds of JReferences like JGlobalReference and JWeakGlobalReference or a global reference that is not managed by a native finalizer.

And it looks like we do want to introduce more types later.